docs/content/modules/carts-and-checkout/backend/add-payment-provider.md

```diff
@@ -1,158 +1,167 @@
-8. `getPaymentData`: Used to retrieve the data that should be stored in the `data` field of a new  Payment instance after the payment amount has been authorized.
-9. `capturePayment`: Used to capture the payment amount of an order or swap.
-10. `refundPayment`: Used to refund a payment amount of an order or swap.
-11. `cancelPayment`: Used to perform any necessary action with the third-party payment provider when an order or swap is canceled.
+5. `deletePayment`: Used to perform any action necessary before a Payment Session is deleted. For example, you can cancel the payment with the third-party provider.
+6. `authorizePayment`: Used to authorize the payment amount of the cart before the order or swap is created.
+7. `capturePayment`: Used to capture the payment amount of an order or swap.
+8. `refundPayment`: Used to refund a payment amount of an order or swap.
+9. `cancelPayment`: Used to perform any necessary action with the third-party payment provider when an order or swap is canceled.
 
 :::note
 
-All these methods must be declared async in the Payment Provider Service.
+All these methods must be declared async in the Payment Processor.
 
 :::
 
 These methods are used at different points in the Checkout flow as well as when processing the order after it’s placed.
 
-![Checkout Flow - Payment](https://res.cloudinary.com/dza7lstvk/image/upload/v1677770739/Medusa%20Docs/Diagrams/checkout-payment_rouet2.png)
+![Checkout Flow - Payment](https://res.cloudinary.com/dza7lstvk/image/upload/v1668001750/Medusa%20Docs/Diagrams/WeDr0ph_idcrir.jpg)
 
 ---
 
-## Create a Payment Provider
+## Create a Payment Processor
 
-The first step to create a payment provider is to create a JavaScript or TypeScript file in `src/services`. The file's name should be the name of the payment provider.
+The first step to create a payment processor is to create a JavaScript or TypeScript file in `src/services`. The file's name should be the name of the payment processor.
 
 For example, create the file `src/services/my-payment.ts` with the following content:
 
 ```ts title=src/services/my-payment.ts
 import { 
-  AbstractPaymentService, PaymentContext, Data, 
-  Payment, PaymentSession, PaymentSessionStatus, 
-  PaymentSessionData, Cart, PaymentData,
-  PaymentSessionResponse } from "@medusajs/medusa"
-
-import { EntityManager } from "typeorm"
+  AbstractPaymentProcessor, 
+  PaymentProcessorContext, 
+  PaymentProcessorError, 
+  PaymentProcessorSessionResponse, 
+  PaymentSessionStatus,
+} from "@medusajs/medusa"
 
-class MyPaymentService extends AbstractPaymentService {
-  protected manager_: EntityManager
-  protected transactionManager_: EntityManager | undefined
+class MyPaymentProcessor extends AbstractPaymentProcessor {
+  static identifier = "my-payment"
 
-  async getPaymentData(
-    paymentSession: PaymentSession
-  ): Promise<Data> {
+  async capturePayment(
```

Comment:
suggestion: Should we order the method like, getStatus, initiate, authorize, capture etc just to give more sense to the flow as well?